File: src/main/java/net/evenh/coffeebot/machine/MachineStatus.java
package net.evenh.coffeebot.machine;

import java.util.Optional;

public enum MachineStatus {
  UP("Up", "good"),
  DOWN("Down", "danger"),
  UNSTABLE("Unstable", "warning");

  public final String readableStatus;
  public final String color;

  MachineStatus(String readableStatus, String color) {
    this.readableStatus = readableStatus;
    this.color = color;
  }

  /**
   * Reverse lookup.
   */
  public static Optional<MachineStatus> fromReadableStatus(String text) {
    for (MachineStatus status : values()) {
      if (status.readableStatus.equalsIgnoreCase(text)) {
        return Optional.of(status);
      }
    }

    return Optional.empty();
  }
}

File: src/main/java/net/evenh/coffeebot/machine/MachineStatus.kt
package net.evenh.coffeebot.machine

import java.util.*

enum class MachineStatus(val readableStatus: String, val color: String) {
    UP("Up", "good"),
    DOWN("Down", "danger"),
    UNSTABLE("Unstable", "warning");

    companion object {
        /**
         * Reverse lookup.
         */
        fun fromReadableStatus(text: String): Optional<MachineStatus> {
            return values()
                    .firstOrNull { it.readableStatus == text.capitalize() }
                    ?.let { Optional.of(it) }
                    ?: Optional.empty()
        }
    }
}
